src/Common/MemoryTracker.cpp

```diff
@@ -353,6 +354,19 @@ AllocationTrace MemoryTracker::allocImpl(Int64 size, bool throw_if_memory_exceed
 
                 /// Prevent recursion. Exception::ctor -> std::string -> new[] -> MemoryTracker::alloc
                 MemoryTrackerBlockerInThread untrack_lock(VariableContext::Global);
+
+#if USE_JEMALLOC
+                if (level == VariableContext::Global && jemalloc_flush_profile_on_memory_exceeded && DB::getJemallocValue<bool>("prof.active"))
```

Comment:
We are doing this only under `can throw` condition, I think we can move it into the outer scope, to make sure we can dump with any new/malloc (as well now)